src/client/pages/myRecommendations/MyRecommendations.jsx

```diff
@@ -96,24 +96,26 @@ const MyRecommendations = () => {
     const [displayConfirm, setDisplayConfirm] = useState(false)
     const [selectedBook, setSelectedBook] = useState(null)
 
+    const remove = (item, list) => {
+        const removeIndex = list.indexOf(item)
+        const deleteCount = 1
+        const newList = [...list]
+        newList.splice(removeIndex, deleteCount)
+        return newList
+    }
+
     const removeTag = (e) => {
         e.preventDefault()
         const removeItem = e.target.className
-        const removeIndex = tags.indexOf(removeItem)
-        const deleteCount = 1
-        const newTags = [...tags]
-        newTags.splice(removeIndex, deleteCount)
-        setTags(newTags)
+        const newList = remove(removeItem, tags)
```

Comment:
I would personally not have called it removeItem in both functions because it's not 100% clear and you're calling it twice with the same parameter name. But it's fine. Not a big deal.